controller/input.cpp

```diff
@@ -163,7 +169,7 @@ void InputDevice::handleFeedbackUpload(uint32_t id)
         return;
     }
 
-    effect = upload.effect;
+    effects[upload.effect.id] = upload.effect; //should really check the id first but the driver always seems to assign the first available from 0.
```

Comment:
This probably should not be done without bounds checking... Especially since the array has a static size. You could look at kernel's ff-memless to look how many effect slots it supports. I believe it was 16. That way, both would behave the same.

But I also think that ff-memless combines effects with each other. With this implementation, only one effect at a time could play and would override the others.